To conforms to RFC822, subject have to be encoded

hykw opened the pull request:
Bamboo.SMTPAdapter.add_subject/2 addes the subject as is, but it have to be encoded to conform to RFC822. It corrupts in some MUA, like Thunderbird, in case of no encodings.

kdisneur commented:
Thank you @hykw for the pull request.

Could you fix the failing tests / add tests before we can merge the Pull Request?

That's the current `mix test` output:

```
  1) test emails looks fine when only text body is set (Bamboo.SMTPAdapterTest)
     test/lib/bamboo/adapters/smtp_adapter_test.exs:168
     Expected truthy, got false
     code: String.contains?(raw_email, "Subject: #{bamboo_email.subject()}\r\n")
     stacktrace:
       test/lib/bamboo/adapters/smtp_adapter_test.exs:187

.

  2) test emails looks fine when only HTML body is set (Bamboo.SMTPAdapterTest)
     test/lib/bamboo/adapters/smtp_adapter_test.exs:207
     Expected truthy, got false
     code: String.contains?(raw_email, "Subject: #{bamboo_email.subject()}\r\n")
     stacktrace:
       test/lib/bamboo/adapters/smtp_adapter_test.exs:226

..

  3) test emails looks fine when text and HTML bodys are sets (Bamboo.SMTPAdapterTest)
     test/lib/bamboo/adapters/smtp_adapter_test.exs:246
     Expected truthy, got false
     code: String.contains?(raw_email, "Subject: #{bamboo_email.subject()}\r\n")
     stacktrace:
       test/lib/bamboo/adapters/smtp_adapter_test.exs:265

.

Finished in 0.3 seconds (0.3s on load, 0.00s on tests)
15 tests, 3 failures
```

Thanks again for your contribution

hykw commented:
I've fixed and added a test to pass it. Because it's a bit difficult for me to write a correct English, you have to correct the comment and the description in test(Sorry for bothering you!)

```
 $ mix test
................

Finished in 0.5 seconds
16 tests, 0 failures
```

tgautier commented:
Tests are passing (we're having a problem with TravisCI and CircleCI).

Thanks a lot @hykw
